client/src/graphics/window.rs

```diff
@@ -222,6 +224,12 @@ impl Window {
                         VirtualKeyCode::F => {
                             down = state == ElementState::Pressed;
                         }
+                        VirtualKeyCode::Space => {
+                            if !jump {
+                                self.sim.set_jump_pressed_true();
```

Comment:
Is it intended that this be called when the key is released?

Reply:
It is not. Good catch.

Jumping is set up so that you keep jumping while holding space, and you jump at least once when you press space regardless of framerate. This logic ends up hiding the bug you noticed.